IBAnimatable/AnimatableImageView.swift

```diff
@@ -110,16 +110,24 @@ import UIKit
   
   public override func layoutSubviews() {
     super.layoutSubviews()
-    
     autoRunAnimation()
   }
   
+  // MARK: - CALayer
+  public override class func layerClass() -> AnyClass {
+    return CAGradientLayer.self
+  }
+  
+  var gradientLayer: CAGradientLayer {
```

Comment:
I completely agree with all the points. At the beginning, I was looking for a better implementation which could answer to that points.

However, I'm not sure I get what do you mean by making `DesignableGradientView` encapsulating all the others subviews. Something like creating a `DesignableGradientView` and adding it at the subviews of the view that conform to `GradientDesignable`? If so, I don't really like that solution since it will make a lot of boilerplate code in each subviews. I really like the fact that it's really easy to make  a new view conform a protocol without that kind of code.

For the moment, I removed all the layer stuff from the views that conform `GradientDesignable`, and moved it directly in the protocol extension. It's clearly not enough since it won't handle the rotation... but I'm gonna think how to improve that.